### ISSUE TITLE
VSCode tooltips do not show default values of pydantic classes.

### ISSUE THREAD

**Issue by user1584** [pydantic/pydantic#10258]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

In VSCode, the tooltip does not show default values:
![PydanticDefaults](https://github.com/user-attachments/assets/d77730bc-ddeb-4b55-a7fe-a0bd7098019e)
I would have expected a similar Tooltip as for regular classes:
![WithoutPydanticDefaults](https://github.com/user-attachments/assets/de279e11-1915-4a9f-89a9-15aa69f4b964)

I'm not sure where the source of the problem is located (pydantic or VSCode or something else) or even if you consider it a bug at all. 

### Example Code

```Python
import pydantic

class Test(pydantic.BaseModel):
    text: str = "hello world"

Test()

class TestWithoutPydantic:
    def __init__(self, text: str = "hello world")
        self.text = text

TestWithoutPydantic()
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.8.2
        pydantic-core version: 2.20.1
          pydantic-core build: profile=release pgo=true
                 install path: /stackable/.local/lib/python3.11/site-packages/pydantic
               python version: 3.11.5 (main, Sep 22 2023, 15:34:29) [GCC 8.5.0 20210514 (Red Hat 8.5.0-20)]
                     platform: Linux-5.15.0-118-generic-x86_64-with-glibc2.28
             related packages: mypy-1.11.1 typing_extensions-4.12.2 pydantic-settings-2.4.0
                       commit: unknown
```

**Comment by Viicos** [comment]
This is a regression in pyright `1.1.372`, which was addressed in `1.1.377`. The related issue can be found here: https://github.com/microsoft/pyright/issues/8781.

You'll have to wait for [Pylance](https://github.com/microsoft/pylance-release/) to ship a new version, or switch to the latest pre release.

**Comment by user1584** [comment]
Thank you!
